File: samples/SelfAspNetVb/SelfAspNetVbIdentity/Chap10/ProfileLabel.vb
Imports System.ComponentModel
Imports Microsoft.AspNet.Identity
Imports Microsoft.AspNet.Identity.Owin

<Designer(GetType(System.Web.UI.Design.ControlDesigner))>
<DefaultProperty("ProfileName")>
<ToolboxData("<{0}:ProfileLabel runat=server></{0}:ProfileLabel>")>
Public Class ProfileLabel
    Inherits Label

    <Bindable(True)>
    <Category("Behavior")>
    <DefaultValue("")>
    <Localizable(True)>
    <Description("プロファイル名")>
    Public Property ProfileName() As String
        Get
            Dim s As String = CStr(ViewState("ProfileName"))
            If s Is Nothing Then
                Return String.Empty
            Else
                Return s
            End If
        End Get

        Set(ByVal Value As String)
            ViewState("ProfileName") = Value
        End Set
    End Property

    <Browsable(False)>
    Public Overrides Property Text() As String
        Get
            Return ""
        End Get
        Set(ByVal value As String)
            Throw New NotSupportedException
        End Set
    End Property

    Protected Overrides Sub RenderContents(ByVal writer As HtmlTextWriter)
        If Context IsNot Nothing Then
            Dim manager = Context.GetOwinContext().GetUserManager(Of ApplicationUserManager)()
            Dim user = manager.FindByName(Context.User.Identity.Name)
            writer.Write(GetType(ApplicationUser).GetProperty(ProfileName).GetValue(user))
        Else
            writer.Write(String.Format("[{0}]", Me.ID))
        End If
    End Sub

End Class

File: samples/SelfAspNet/SelfAspNetIdentity/Chap10/ProfileLabel.cs
using Microsoft.AspNet.Identity;
using Microsoft.AspNet.Identity.Owin;
using SelfAspNetIdentity.Models;
using System;
using System.ComponentModel;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace SelfAspNetIdentity.Chap10
{
    [Designer(typeof(System.Web.UI.Design.ControlDesigner))]
    [DefaultProperty("ProfileName")]
    [ToolboxData("<{0}:ProfileLabel runat=server></{0}:ProfileLabel>")]
    public class ProfileLabel : Label
    {
        [Bindable(true)]
        [Category("Behavior")]
        [DefaultValue("")]
        [Localizable(true)]
        [Description("プロファイル名")]
        public string ProfileName
        {
            get
            {
                var s = (String)ViewState["ProfileName"];
                if (s == null)
                {
                    return String.Empty;
                }
                else
                {
                    return s;
                }
            }

            set
            {
                ViewState["ProfileName"] = value;
            }
        }

        [Browsable(false)]
        public override String Text
        {
            get
            {
                return "";
            }

            set
            {
                throw new NotSupportedException();
            }
        }

        protected override void RenderContents(HtmlTextWriter writer)
        {
            var cx = this.Context;
            if (cx != null)
            {
                var manager = Context.GetOwinContext().GetUserManager<ApplicationUserManager>();
                var user = manager.FindByName(Context.User.Identity.Name);
                writer.Write(typeof(ApplicationUser).GetProperty(ProfileName).GetValue(user));
            }
            else
            {
                writer.Write(String.Format("[{0}]", this.ID));
            }
        }
    }
}
